cypress/integration/field/reviewChanges.test.js

```diff
@@ -0,0 +1,128 @@
+import {uuid} from '@sanity/uuid'
+import client from '../../helpers/sanityClientSetUp'
+
+const DOC_ID = `drafts.${uuid()}`
+
+const doc = {
+  _id: DOC_ID,
+  _type: 'reviewChanges',
+  stringWithDefaultDiffComponent: undefined,
+  stringWithCustomDiffComponent: undefined,
+  stringWithRenderDiffComponent: undefined,
+  numberWithDefaultDiffComponent: undefined,
+  numberWithCustomDiffComponent: undefined,
+  numberWithRenderDiffComponent: undefined,
+  booleanWithDefaultDiffComponent: undefined,
+  booleanWithCustomDiffComponent: undefined,
+  booleanWithRenderDiffComponent: undefined,
+  slugWithDefaultDiffComponent: undefined,
+  slugWithCustomDiffComponent: undefined,
+  slugWithRenderDiffComponent: undefined,
+}
+
+function patchDocument(payload) {
+  return client.patch(DOC_ID).set(payload).commit({visibility: 'async'})
+}
+
+describe('@sanity/field: Review changes', () => {
+  before(async () => {
+    await client.create(doc, {visibility: 'async'})
+  })
+
+  beforeEach(() => {
+    cy.visit(
+      `/test/content/input-ci;reviewChanges;${DOC_ID}%2Ctemplate%3DreviewChanges%2Csince%3D%40lastPublished`
+    )
+    cy.viewport(1500, 3500)
+  })
+
+  it('panel is visible', () => {
+    cy.get('[data-testid="review-changes-pane"]').should('be.visible')
+  })
+
+  it('no changes message visible', () => {
+    cy.get('[data-testid="review-changes-no-changes-message"]').should('be.visible')
+  })
+
+  it('default diff is visible', () => {
+    cy.wrap(
+      patchDocument({
+        booleanWithDefaultDiffComponent: true,
+        numberWithDefaultDiffComponent: 1,
+        slugWithDefaultDiffComponent: {
+          current: 'slug',
+        },
+        stringWithDefaultDiffComponent: 'Hello world',
+      })
+    ).then(() => {
+      cy.get('[data-testid="string-field-diff"]').should('be.visible')
+      cy.get('[data-testid="boolean-field-diff"]').should('be.visible')
+      cy.get('[data-testid="slug-field-diff"]').should('be.visible')
+      cy.get('[data-testid="number-field-diff"]').should('be.visible')
+    })
+  })
+
+  // Configured in the reviewChanges schema
+  it('components.diff is visible', () => {
```

Comment:
Hm, there is potential for race conditions between these tests. Since patching happens in multiple tests at the same time, there is potential for data to interfered between tests.

I think we should

Reply:
Are you sure? Because a new document is created for each test

Reply:
I think my comment was posted before you updated your comment :) 

What I meant was: A unique document is already created in this test, and later deleted once it's completed. But, if I understand you correctly, you think we should create several documents for each case in this test?

Reply:
Well, one document for each test should be sufficient 👍 

The reason is that jest runs tests in parallel – so we need to make sure the test code supports that.